### PR TITLE
Fix bugs in a couple of Lua nk module functions.

### PR DESCRIPTION
**Pull request by sesposito**
1) jwt_generate now correctly supports the RS256 algorithm.
2) rsa_sha256_hash now correctly decoding the private key argument.

### CODE REVIEW

File: server/runtime_lua_nakama.go [hunk 1]
```diff
@@ -1129,7 +1136,8 @@ func (n *RuntimeLuaNakamaModule) rsaSHA256Hash(l *lua.LState) int {
 		return 0
 	}
 
-	rsaPrivateKey, err := x509.ParsePKCS1PrivateKey([]byte(key))
+	block, _ := pem.Decode([]byte(key))
```
Comment (zyro) [inline server/runtime_lua_nakama.go:1139]: We should observe the error here and return early if non-nil.

**Review by zyro — CHANGES_REQUESTED** [review]

**Comment by zyro** [comment]
Also in the algo selection here https://github.com/heroiclabs/nakama/pull/457/files#diff-ac71981a54f4a4b70786ee656dde0bcfR811 the default case should be an arg error for the 1st parameter (not the 3rd) and we should also return early there.

**Comment by zyro** [comment]
@sesposito Couple of small changes otherwise looks nice! 👍

File: server/runtime_lua_nakama.go (hunk 1 above)
Comment (sesposito) [reply server/runtime_lua_nakama.go:1139]: The second returned value of `pem.Decode` is not an error, but the remaining input if it has multiple pem formatted blocks. I'll check if the returned `block` is nil instead and raise an error if it is.

Comment (zyro) [reply server/runtime_lua_nakama.go:1139]: Excellent point, the check if it's `nil` is a good move then.

**Review by zyro — APPROVED** [review]

**Pull request merged**